build_tools/update_environments_and_lock_files.py

```diff
@@ -367,6 +367,13 @@ def remove_from(alist, to_remove):
         ],
         "package_constraints": {
             "python": "3.9",
+            # TODO: this needs to be adapted when matplotlib 3.11 is out. In
+            # the meantime, this avoids a warning in matplotlib 3.9 boxplot
+            # labels has been renamed to tick_labels. Possible options:
+            # - bump minimum matplotlib supported versions to 3.9 at one point
+            # - complicate the example code to do the right thing depending on
+            #   maplotlib version
+            "matplotlib": "<3.9",
```

Comment:
what about fixing the code, so that our code actually runs warning free for people with an up to date env, and ignore those warnings in the min-version of the CI?

Reply:
> And I think @glemaitre and I are switching triage duty since he's on vacation next week

Right, I likely missed the update :wink:

> what about fixing the code, so that our code actually runs warning free for people with an up to date env, and ignore those warnings in the min-version of the CI?

Fixing the code is the "complicate the example code" option (if matplotlib recent enough use the new way else use the old way) I tried to describe in the comment, maybe let's leave this for a separate PR?

For completeness, this is a DeprecationWarning in matplotlib:
```
matplotlib._api.deprecation.MatplotlibDeprecationWarning: The 'labels' parameter of boxplot() has been renamed 'tick_labels' since Matplotlib 3.9; support for the old name will be dropped in 3.11.
```

3 examples fail (DeprecationWarning turned into error):
- examples/inspection/plot_permutation_importance_multicollinear.py
- examples/ensemble/plot_gradient_boosting_regression.py failed
- examples/release_highlights/plot_release_highlights_0_22_0.py

See [build log](https://app.circleci.com/pipelines/github/scikit-learn/scikit-learn/59445/workflows/2fe8c71c-c157-4882-b386-59028fb62f15/jobs/278637) from one of the commit in this PR.

Reply:
I opened https://github.com/scikit-learn/scikit-learn/issues/29434 to not forget. In practice, one day or another I would have bumped into this TODO and be annoyed enough to do something about it :wink:.